### ISSUE TITLE
Store if a surface has been modified (1434)

### ISSUE THREAD

**Issue by GalacticEmperor1** [pygame-community/pygame-ce#811]
#  **Issue &numero;1434 opened by *[robertpfeiffer](https://github.com/robertpfeiffer)* at 2019-10-18 11:16:36**

Putting this up for discussion:

Can we store a flag that tracks whether a surface has been modified, so that every blit, draw, or set_at operation sets that flag to `True`, with methods to retrieve or clear that flag? This might be very useful for writing code that copies textures to the GPU (SDL_Renderer/GL) and updates them only when they have been changed.

**Related Docs:** https://www.pygame.org/docs/ref/surface.html<hr>

#  Comments

# # #  **[illume](https://github.com/illume)* commented at 2019-10-18 21:51:00*

Yeah, I guess this is possible.

Won't be able to catch all changes, but probably many.
The hard one will be detecting changes in a buffer/array, or shared memory situation.
<hr>

# # #  **[humanbeast7](https://github.com/humanbeast7)* commented at 2019-10-21 11:53:02*

to me, it's sounds like you suggested turning all surfaces to Dirty Sprites with less features
which will make Sprites and non-dirty Groups irrelevant
<hr>

# # #  **[robertpfeiffer](https://github.com/robertpfeiffer)* commented at 2019-10-21 17:58:22*

> to me, it's sounds like you suggested turning all surfaces to Dirty Sprites with less features
> which will make Sprites and non-dirty Groups irrelevant

Basically, I'd like to know if we can make DirtySprites automatically dirty.

**Comment by Starbuck5** [comment]
This is a cool idea and could be useful, but as mentioned we won't be able to catch changes in a buffer or shared memory situation.

And it has been dormant for a while, so I'm going to close as not planned.
